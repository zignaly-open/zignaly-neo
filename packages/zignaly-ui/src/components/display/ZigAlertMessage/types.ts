export type ZigAlertMessageProps = {
  text: string | React.ReactElement;
  error?: boolean;
  warning?: boolean;
  id?: string;
};
