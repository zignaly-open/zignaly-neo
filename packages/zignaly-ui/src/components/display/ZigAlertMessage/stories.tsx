import * as React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import ZigAlertMessage from "./";

export default {
  title: "Display/ZigAlertMessage",
  component: ZigAlertMessage,
  argTypes: {
    text: {
      type: "string",
    },
    error: {
      defaultValue: true,
      type: "boolean",
    },
    warning: {
      type: "boolean",
    },
  },
} as ComponentMeta<typeof ZigAlertMessage>;

const Template: ComponentStory<typeof ZigAlertMessage> = (args) => <ZigAlertMessage {...args} />;

export const Error = Template.bind({});
Error.args = {
  text: "Exceeds available balance",
  error: true,
};
