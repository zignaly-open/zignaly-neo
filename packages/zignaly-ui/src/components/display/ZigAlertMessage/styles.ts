import styled from "styled-components";

export const Layout = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const Icon = styled.div`
  margin-right: 8px;
  display: flex;
  align-items: center;
`;
