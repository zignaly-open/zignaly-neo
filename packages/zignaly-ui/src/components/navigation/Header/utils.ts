// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
