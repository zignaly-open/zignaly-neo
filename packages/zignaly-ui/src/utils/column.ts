import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
