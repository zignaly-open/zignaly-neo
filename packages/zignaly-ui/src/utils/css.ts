import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
