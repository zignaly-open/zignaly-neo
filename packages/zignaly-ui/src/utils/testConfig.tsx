import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
