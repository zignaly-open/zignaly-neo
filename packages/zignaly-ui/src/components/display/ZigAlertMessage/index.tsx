import React from "react";

import { Layout, Icon } from "./styles";

import { ZigAlertMessageProps } from "./types";

import { ReactComponent as ErrorAlertIcon } from "assets/icons/error-alert-icon.svg";
import { dark } from "theme";
import ZigTypography from "../ZigTypography";

// TODO(Vetle): Refactor usage of Dark theme when light theme is ready
function ZigAlertMessage({ text, error, warning, id }: ZigAlertMessageProps) {
  return (
    <Layout id={id}>
      <Icon>
        <ErrorAlertIcon
          height="24px"
          width="24px"
          color={error ? dark.redGraphOrError : warning ? dark.yellow : dark.neutral400}
        />
      </Icon>
      <ZigTypography
        variant="body2"
        sx={{ color: error ? dark.redGraphOrError : warning ? dark.yellow : dark.neutral400 }}
      >
        {text}
      </ZigTypography>
    </Layout>
  );
}

export default ZigAlertMessage;

// added for backwards compatibility
export const ErrorMessage: React.FC<Pick<ZigAlertMessageProps, "text" | "id">> = ({ text, id }) => (
  <ZigAlertMessage text={text} error id={id} />
);
