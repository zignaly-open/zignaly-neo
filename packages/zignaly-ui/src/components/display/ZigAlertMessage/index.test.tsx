import React from "react";
import ZigAlertMessage, { ErrorMessage } from "./index";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("components/display/ZigAlertMessage", () => {
  it("renders as an alert", async () => {
    const { container } = renderWithProvidersUi(
      <ZigAlertMessage text={"test text"} id={"alert"} />,
    );

    const alert = container.querySelector("#alert") as Element;
    expect(alert).toBeVisible();
    const alertStyles = getComputedStyle(alert);
    expect(alert).toMatchSnapshot();
    expect(alertStyles).toMatchSnapshot();
  });

  it("renders as a warning", async () => {
    const { container } = renderWithProvidersUi(
      <ZigAlertMessage warning text={"test text"} id={"alert"} />,
    );

    const alert = container.querySelector("#alert") as Element;
    expect(alert).toBeVisible();
    const alertStyles = getComputedStyle(alert);
    expect(alert).toMatchSnapshot();
    expect(alertStyles).toMatchSnapshot();
  });

  it("renders as error", async () => {
    const { container } = renderWithProvidersUi(<ErrorMessage text={"test text"} id={"error"} />);

    const error = container.querySelector("#error") as Element;
    expect(error).toBeVisible();
    const errorStyles = getComputedStyle(error);
    expect(error).toMatchSnapshot();
    expect(errorStyles).toMatchSnapshot();
  });
});
