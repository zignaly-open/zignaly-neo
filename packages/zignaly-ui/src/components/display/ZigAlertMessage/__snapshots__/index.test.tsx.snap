// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigAlertMessage renders ErrorMessage render 1`] = `
<div
  class="sc-bczRLJ bxlVqq"
  id="error"
>
  <div
    class="sc-gsnTZi dhsoie"
  >
    <svg
      color="#CC3993"
      height="24px"
      width="24px"
    />
  </div>
  <span
    class="MuiTypography-root MuiTypography-body2 css-1wlkhe7-MuiTypography-root"
  >
    test text
  </span>
</div>
`;

exports[`components/display/ZigAlertMessage renders ErrorMessage render 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "2": "align-items",
  "3": "visibility",
  "_importants": {
    "align-items": "",
    "display": "",
    "flex-direction": "",
    "visibility": undefined,
  },
  "_length": 4,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "display": "flex",
    "flex-direction": "row",
    "visibility": "visible",
  },
}
`;

exports[`components/display/ZigAlertMessage renders ZigAlertMessage render 1`] = `
<div
  class="sc-bczRLJ bxlVqq"
  id="alert"
>
  <div
    class="sc-gsnTZi dhsoie"
  >
    <svg
      color="#706F82"
      height="24px"
      width="24px"
    />
  </div>
  <span
    class="MuiTypography-root MuiTypography-body2 css-xsotb5-MuiTypography-root"
  >
    test text
  </span>
</div>
`;

exports[`components/display/ZigAlertMessage renders ZigAlertMessage render 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "2": "align-items",
  "3": "visibility",
  "_importants": {
    "align-items": "",
    "display": "",
    "flex-direction": "",
    "visibility": undefined,
  },
  "_length": 4,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "display": "flex",
    "flex-direction": "row",
    "visibility": "visible",
  },
}
`;

exports[`components/display/ZigAlertMessage renders ZigAlertMessage render warning 1`] = `
<div
  class="sc-bczRLJ bxlVqq"
  id="alert"
>
  <div
    class="sc-gsnTZi dhsoie"
  >
    <svg
      color="#AE9C5B"
      height="24px"
      width="24px"
    />
  </div>
  <span
    class="MuiTypography-root MuiTypography-body2 css-13jpils-MuiTypography-root"
  >
    test text
  </span>
</div>
`;

exports[`components/display/ZigAlertMessage renders ZigAlertMessage render warning 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "flex-direction",
  "2": "align-items",
  "3": "visibility",
  "_importants": {
    "align-items": "",
    "display": "",
    "flex-direction": "",
    "visibility": undefined,
  },
  "_length": 4,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "display": "flex",
    "flex-direction": "row",
    "visibility": "visible",
  },
}
`;
